### PR TITLE
Add types for P6 GenesisData

### PR DESCRIPTION
**Pull request by limemloh**
## Purpose

Introduces changes needed for genesis-creator to support P6
Related to https://github.com/Concordium/concordium-misc-tools/pull/51

## Changes

- Implement `Serial` for `num::rational::Ratio<u64>` and `Duration`.
- Introduce types for protocol version 6: `ChainParameterVersion2`, `GASRewardsCPV1`, `TimeoutParameters` and `FinalizationCommitteeParameters`.

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: rust-src/concordium_base/src/base.rs [hunk 1]
```diff
@@ -1145,8 +1147,30 @@ impl MintDistributionFamily for ChainParameterVersion1 {
     type Output = MintDistributionV1;
 }
 
+impl MintDistributionFamily for ChainParameterVersion2 {
+    type Output = MintDistributionV1;
+}
+
 pub type MintDistribution<CPV> = <CPV as MintDistributionFamily>::Output;
 
+pub trait GASRewardsFamily {
```
Comment (abizjak) [inline rust-src/concordium_base/src/base.rs:1156]: Please add a bit of documentation.

File: rust-src/concordium_base/src/common/serialize.rs [hunk 2]
```diff
@@ -435,6 +436,15 @@ impl Serial for std::num::NonZeroI128 {
     }
 }
 
+impl Serial for num::rational::Ratio<u64> {
+    fn serial<B: Buffer>(&self, out: &mut B) {
+        out.write_u64::<BigEndian>(*self.numer())
+            .expect("Writing to a buffer should not fail.");
```
Comment (abizjak) [inline rust-src/concordium_base/src/common/serialize.rs:442]:
```suggestion
        self.numer().serial(out);
```

File: rust-src/concordium_base/src/common/serialize.rs [hunk 3]
```diff
@@ -435,6 +436,15 @@ impl Serial for std::num::NonZeroI128 {
...
+        out.write_u64::<BigEndian>(*self.denom().deref())
+            .expect("Writing to a buffer should not fail.");
```
Comment (abizjak) [inline rust-src/concordium_base/src/common/serialize.rs:444]:
```suggestion
        self.denom().deref().serial(out);
```

File: rust-src/concordium_base/src/updates.rs [hunk 4]
```diff
@@ -522,6 +555,22 @@ pub struct PoolParameters {
...
+/// in the finalization committee.
+pub struct FinalizationCommitteeParameters {
+    /// Minimum number of bakers to include in the finalization committee before
+    /// the '_fcpFinalizerRelativeStakeThreshold' takes effect.
+    pub min_finalizers: u32,
+    /// Maximum number of bakers to include in the finalization committee.
+    pub max_finalizers: u32,
+    /// Determining the staking threshold required for being eligible the
+    /// finalization committee. The required amount is given by `total stake
+    /// in pools * finalizer_relative_stake_threshold` Provided as part per
```
Comment (abizjak) [inline rust-src/concordium_base/src/updates.rs:569]:
```suggestion
    /// in pools * finalizer_relative_stake_threshold` provided as part per
```

File: rust-src/concordium_base/src/updates.rs [hunk 5]
```diff
@@ -522,6 +555,22 @@ pub struct PoolParameters {
     pub leverage_bound:                  LeverageFactor,
 }
 
+#[derive(Debug, common::Serial, Clone)]
```
Comment (abizjak) [inline rust-src/concordium_base/src/updates.rs:558]: Why not Copy?

File: rust-src/concordium_base/CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,8 @@
 ## Unreleased changes
 
+- Implement Serial for `num::rational::Ratio<u64>` and `Duration`.
```
Comment (abizjak) [inline rust-src/concordium_base/CHANGELOG.md:3]: Please add Deserial here as well when you implement.

**Review by abizjak — APPROVED** [review]

File: rust-src/concordium_base/src/updates.rs [hunk 7]
```diff
@@ -522,6 +582,22 @@ pub struct PoolParameters {
     pub leverage_bound:                  LeverageFactor,
 }
 
+#[derive(Debug, common::Serialize, Clone, Copy)]
+/// Finalization committee parameters. These parameters control which bakers are
+/// in the finalization committee.
+pub struct FinalizationCommitteeParameters {
+    /// Minimum number of bakers to include in the finalization committee before
+    /// the '_fcpFinalizerRelativeStakeThreshold' takes effect.
```
Comment (td202) [inline rust-src/concordium_base/src/updates.rs:590]:
```suggestion
    /// the `finalizers_relative_stake_threshold` takes effect.
```

**Review by td202 — APPROVED** [review]

**Pull request merged**